### PR TITLE
Minor bug fixes

### LINKED ISSUE #513: Check for None before Calling validate_has_key
[This code](https://github.com/Azure/WALinuxAgent/blob/master/azurelinuxagent/ga/exthandlers.py#L73-L75) should check for None before calling validate_has_key.  The following stack trace is from an older agent, and show the method validate_has_key failing due to a None object.

```text
2016/12/13 19:05:55.602102 ERROR Traceback (most recent call last):
  File "/usr/sbin/waagent", line 35, in <module>
    agent.main()
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/agent.py", line 141, in main
    run()
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/agent.py", line 46, in run
    HANDLERS.main_handler.run()
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/run.py", line 70, in run
    h_status_list = self.handlers.extension_handler.process()
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 112, in process
    h_status = self.process_extension(extension, pkg_list)
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 128, in process_extension
    status = ext.collect_handler_status()
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 371, in collect_handler_status
    ext_status = self.collect_extension_status()
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 389, in collect_extension_status
    self.settings.sequenceNumber)
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 99, in ext_status_to_v2
    status.substatusList.extend(extension_sub_status_to_v2(substatus))
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 57, in extension_sub_status_to_v2
    'substatus/formattedMessage/lang')
  File "/usr/lib/python3.4/dist-packages/azurelinuxagent/distro/default/extension.py", line 37, in validate_has_key
    if key not in obj:
TypeError: argument of type 'NoneType' is not iterable

2016/12/13 19:05:56.722887 INFO Azure Linux Agent Version:2.1.1
2016/12/13 19:05:56.735031 INFO OS: ubuntu 15.10
```

This is an older version, and deprecated OS, but I believe the problem is still valid.  There may be other places in the code that I missed as well.

This is also causing the agent to go into a death spiral.  It is constantly crashing and restarting, and the logs are growing quite fast.  I suspect this is due to a bug in the VMSnapshotLinux extension.

### PR DESCRIPTION
**Pull request by hglkrijger**
- reduce httplib logging 
- add missing None check
- correct exception type handling
- fixes #513, #519, #528 

/cc @jinhyunr @brendandixon

### CODE REVIEW

File: azurelinuxagent/common/utils/restutil.py [hunk 1]
```diff
@@ -112,15 +114,19 @@ def http_request(method, url, data, headers=None, max_retry=3,
 
     # If httplib module is not built with ssl support. Fallback to http
     if secure and not hasattr(httpclient, "HTTPSConnection"):
-        logger.warn("httplib is not built with ssl support")
         secure = False
+        if secure_warning:
+            logger.warn("httplib is not built with ssl support")
+            secure_warning = False
```
Comment (brendandixon) [inline azurelinuxagent/common/utils/restutil.py]: @hglkrijger I can see you set it `False` here, but where does it ever achieve `True`?

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -107,7 +107,8 @@ def parse_ext_status(ext_status, data):
     if substatus_list is None:
         return
     for substatus in substatus_list:
-        ext_status.substatusList.append(parse_ext_substatus(substatus))
+        if substatus is not None:
```
Comment (brendandixon) [inline azurelinuxagent/ga/exthandlers.py]: @hglkrijger Would a better check be to see if `ext_status.substatuslist` is a list (that is, check for vs. check absence)?

File: azurelinuxagent/common/utils/restutil.py (hunk 1 above)
Comment (hglkrijger) [inline azurelinuxagent/common/utils/restutil.py]: line 32

File: azurelinuxagent/ga/exthandlers.py (hunk 2 above)
Comment (hglkrijger) [inline azurelinuxagent/ga/exthandlers.py]: The issue we see is that `substatus` is `None`, and the parse in line 111 fails.

File: azurelinuxagent/common/utils/restutil.py (hunk 1 above)
Comment (brendandixon) [inline azurelinuxagent/common/utils/restutil.py]: @hglkrijger Hmm, how did I miss that? :P

File: azurelinuxagent/ga/exthandlers.py (hunk 2 above)
Comment (brendandixon) [inline azurelinuxagent/ga/exthandlers.py]: @hglkrijger I understand. But 111 needs a list...so why not check (if possible) that `ext_status.substatusList` is a list (vs. simply ensuring it's not `None`)?

Comment (brendandixon) [inline azurelinuxagent/ga/exthandlers.py]: @hglkrijger Nevermind. Senility.

**Pull request merged**